src/basic/namespace-util.c

```diff
@@ -527,7 +527,7 @@ int is_idmapping_supported(const char *path) {
                 return r;
 
         userns_fd = userns_acquire(uid_map, gid_map);
-        if (ERRNO_IS_NEG_NOT_SUPPORTED(userns_fd))
+        if (ERRNO_IS_NEG_NOT_SUPPORTED(userns_fd) || userns_fd == -EPERM)
```

Comment:
ERRNO_IS_NEG_PRIVILEGE() ?

Reply:
you might also want to add that to the open_tree() handling below

Reply:
btw, the open() call below should never fail on EINVAL, that error code should really be dropped there.

Reply:
And the EPERM check on mount_setattr() should also be ERRNO_IS_NEG_PRIVILEGE()